lib/teleterm/gateway/config.go

```diff
@@ -123,15 +123,15 @@ func (c *Config) CheckAndSetDefaults() error {
 	}
 
 	c.Log = c.Log.WithFields(logrus.Fields{
-		"resource": c.TargetURI,
+		"resource": c.TargetURI.String(),
 		"gateway":  c.URI.String(),
 	})
 
 	if c.TargetName == "" {
 		return trace.BadParameter("missing target name")
 	}
 
-	if c.TargetURI == "" {
+	if c.TargetURI.String() == "" {
```

Comment:
I think we should move this fragment above the part with `c.Log =` so that we validate `c.TargetURI` before we actually attempt to call `String` on it.

It probably makes no difference when it comes to the final behavior but I feel like it makes more sense when it comes to code hygiene, idk.